impl/impl.go

```diff
@@ -217,19 +233,33 @@ func (m *manager) OpenPushDataChannel(ctx context.Context, requestTo peer.ID, vo
 
 // OpenPullDataChannel opens a data transfer that will request data from the sending peer and
 // transfer parts of the piece that match the selector
-func (m *manager) OpenPullDataChannel(ctx context.Context, requestTo peer.ID, voucher datatransfer.TypedVoucher, baseCid cid.Cid, selector datamodel.Node) (datatransfer.ChannelID, error) {
+func (m *manager) OpenPullDataChannel(ctx context.Context, requestTo peer.ID, voucher datatransfer.TypedVoucher, baseCid cid.Cid, selector datamodel.Node, options ...datatransfer.TransferOption) (datatransfer.ChannelID, error) {
 	log.Infof("open pull channel to %s with base cid %s", requestTo, baseCid)
 
+	tc := datatransfer.FromOptions(options)
+
 	req, err := m.newRequest(ctx, selector, true, voucher, baseCid, requestTo)
 	if err != nil {
 		return datatransfer.ChannelID{}, err
 	}
+
 	// initiator = us, sender = them, receiver = us
 	chid, err := m.channels.CreateNew(m.peerID, req.TransferID(), baseCid, selector, voucher,
 		m.peerID, requestTo, m.peerID)
 	if err != nil {
 		return chid, err
 	}
+
+	eventsCb := tc.EventsCb()
+	if eventsCb != nil {
+		m.channelSubscriptions.Subscribe(chid, eventsCb)
+	}
+
+	err = m.channels.Open(chid)
+	if err != nil {
+		return chid, err
+	}
```

Comment:
ditto, take it or leave it

```suggestion
	if eventsCb := tc.EventsCb(); eventsCb != nil {
		m.channelSubscriptions.Subscribe(chid, eventsCb)
	}

	if err := m.channels.Open(chid); err != nil {
		return chid, err
	}
```